src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -40,6 +43,8 @@ public class GoblintAnalysis implements ServerAnalysis {
     private final GobPieConfiguration gobpieConfiguration;
     private final FileAlterationObserver goblintConfObserver;
 
+    private Boolean analysisRunning = false;
```

Comment:
Could it be that Magpie runs our `analyze` in different threads in parallel? If that's the case, then `AtomicBoolean` may be necessary.

Reply:
I replaced `analysisRunning` type with `AtomicBoolean` and moved its value assignments near the actual request and response calls. 

I then tested the code a bit more than previously: not only with the first analysis (during the IDE/server session) but also with the second analysis. Both, `boolean` and `AtomicBoolean` variants seemed to work well for aborting the first analysis. But the second analysis would not be aborted when a new analysis is triggered, and instead, the new analysis will wait until the first finishes and then start. I investigated it with a bunch of logging and it seems that the logic with the booleans should actually be correct, but the `analyze` method itself is not called until the previous `analyze` call finishes. So, I am not too sure about how MagpieBridge threads work and why new calls to the `analyze` method seem to run in parallel with the first analysis, but not the following ones 🤷.

Reply:
That's unfortunate, I suppose there could be a work queue/thread pool in Magpie with a limited amount of parallel threads, so once they're all doing something, new tasks will remain waiting in the queue for something else to complete.
That probably requires deeper investigation.